platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/PingAwareSolrClientProxy.java

```diff
@@ -11,36 +11,45 @@
  * License is distributed along with this program and can be found at
  * <http://www.gnu.org/licenses/lgpl.html>.
  */
-package ddf.catalog.cache.solr.impl;
+package org.codice.solr.factory.impl;
 
 import java.io.IOException;
 import org.apache.solr.client.solrj.SolrClient;
 import org.apache.solr.client.solrj.SolrServerException;
-import org.apache.solr.client.solrj.response.UpdateResponse;
+import org.apache.solr.client.solrj.response.SolrPingResponse;
 
 /**
- * Implementation of the {@link SolrClientAdaptor.State} interface that delegates all its operations
- * to an initialized {@link SolrClient}.
+ * This proxy provides the ability to segregate ping requests from other requests by sending them to
+ * a second Solr client which would typically be configured not to perform any internal retries.
  */
-class InitializedSolrClientAdaptor implements SolrClientAdaptor.State {
+public class PingAwareSolrClientProxy extends SolrClientProxy {
   private final SolrClient client;
 
-  InitializedSolrClientAdaptor(SolrClient client) {
+  private final SolrClient pingClient;
+
+  public PingAwareSolrClientProxy(SolrClient client, SolrClient pingClient) {
     this.client = client;
+    this.pingClient = pingClient;
+  }
+
+  @Override
+  protected SolrClient getProxiedClient() {
+    return client;
   }
 
   @Override
-  public void commit() throws SolrServerException, IOException {
-    client.commit();
+  public SolrPingResponse ping() throws SolrServerException, IOException {
+    return pingClient.ping();
   }
 
   @Override
   public void close() throws IOException {
-    client.close();
+    try (final SolrClient c = client;
```

Comment:
❓Do you think we could use the `Closer` here? In the constructor we could do
```
this.client = closer.with(client);
this.pingClient = closer.with(pingCllient);
```

and simply call `closer.close()` here.

Reply:
We could, but why use another object to do what Java can do so nicely like this? Plus Java here will take care of properly throwing back any IOException adding as suppressed exceptions the other if it failed too. The current closer didn't deal with throwing back the IOException as I didn't have a need where it was used.